hamilton/function_modifiers/adapters.py

```diff
@@ -49,9 +50,18 @@ def validate(self):
 
         :raises InvalidDecoratorException: If the arguments are invalid.
         """
+        # get_required_arguments won't catch dataclasses.field(default_factory=...)
```

Comment:
Why not just change it to? Makes more sense here: https://github.com/DAGWorks-Inc/hamilton/blob/72866175bb1e5dc65cedfd2de1096cc082fa19ee/hamilton/io/data_adapters.py#L64.